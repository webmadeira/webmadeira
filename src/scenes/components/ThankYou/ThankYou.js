import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import AppContainer from '../../../components/AppContainer/AppContainer'
import { H2, Body2 } from '../../../components/typography'
import Sponsors from './components/Sponsors/Sponsors'

const Root = styled.section`
  padding-top: 80px;
  padding-bottom: 80px;
`

const Title = styled(H2)`
  text-align: center;
  color: ${props => props.theme.color('brand', 'light')};
  text-transform: uppercase;
  margin-bottom: 40px;
  
  ${({ theme }) => theme.mixins.mediaQuery.phone`
    margin-bottom: 40px;
  `};
`

const Text = styled(Body2)`
  text-align: justify;

  a {
    color: ${props => props.theme.color('neutral', 'base')};
  }
`

const ThankYou = ({ sponsors, thankYouText }) => (
  <AppContainer>
    <Root>
      <Title>{'{ "Thank you" }'}</Title>
      <Text dangerouslySetInnerHTML={{ __html: thankYouText }} />
      <Sponsors sponsors={sponsors} />
    </Root>
  </AppContainer>
)

ThankYou.propTypes = {
  thankYouText: PropTypes.string.isRequired,
  sponsors: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
}

export default ThankYou
