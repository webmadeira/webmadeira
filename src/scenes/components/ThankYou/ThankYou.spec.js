import React from 'react'
import { shallow } from 'enzyme'
import ThankYou from './ThankYou'

describe('ThankYou component', () => {
  it('must render properly', () => {
    const thankYouText = ''
    const sponsors = [
      {
        id: 0,
        name: '',
        logo: '',
        link: '',
      },
    ]

    const component = shallow(<ThankYou sponsors={sponsors} thankYouText={thankYouText} />)

    expect(component).toMatchSnapshot()
  })
})
