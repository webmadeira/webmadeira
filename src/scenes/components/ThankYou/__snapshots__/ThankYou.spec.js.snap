// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThankYou component must render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ThankYou
    sponsors={
      Array [
        Object {
          "id": 0,
          "link": "",
          "logo": "",
          "name": "",
        },
      ]
    }
    thankYouText=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <styled.section>
        <Styled(styled.h2)>
          { "Thank you" }
        </Styled(styled.h2)>
        <Styled(styled.p)
          dangerouslySetInnerHTML={
            Object {
              "__html": "",
            }
          }
        />
        <Sponsors
          sponsors={
            Array [
              Object {
                "id": 0,
                "link": "",
                "logo": "",
                "name": "",
              },
            ]
          }
        />
      </styled.section>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(styled.h2)>
            { "Thank you" }
          </Styled(styled.h2)>,
          <Styled(styled.p)
            dangerouslySetInnerHTML={
              Object {
                "__html": "",
              }
            }
          />,
          <Sponsors
            sponsors={
              Array [
                Object {
                  "id": 0,
                  "link": "",
                  "logo": "",
                  "name": "",
                },
              ]
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "{ \\"Thank you\\" }",
          },
          "ref": null,
          "rendered": "{ \\"Thank you\\" }",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "dangerouslySetInnerHTML": Object {
              "__html": "",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "sponsors": Array [
              Object {
                "id": 0,
                "link": "",
                "logo": "",
                "name": "",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <styled.section>
          <Styled(styled.h2)>
            { "Thank you" }
          </Styled(styled.h2)>
          <Styled(styled.p)
            dangerouslySetInnerHTML={
              Object {
                "__html": "",
              }
            }
          />
          <Sponsors
            sponsors={
              Array [
                Object {
                  "id": 0,
                  "link": "",
                  "logo": "",
                  "name": "",
                },
              ]
            }
          />
        </styled.section>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Styled(styled.h2)>
              { "Thank you" }
            </Styled(styled.h2)>,
            <Styled(styled.p)
              dangerouslySetInnerHTML={
                Object {
                  "__html": "",
                }
              }
            />,
            <Sponsors
              sponsors={
                Array [
                  Object {
                    "id": 0,
                    "link": "",
                    "logo": "",
                    "name": "",
                  },
                ]
              }
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "{ \\"Thank you\\" }",
            },
            "ref": null,
            "rendered": "{ \\"Thank you\\" }",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "dangerouslySetInnerHTML": Object {
                "__html": "",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "sponsors": Array [
                Object {
                  "id": 0,
                  "link": "",
                  "logo": "",
                  "name": "",
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
